lib/http/allowed.ts

```diff
@@ -0,0 +1,37 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { getSession } from "next-auth/client";
+
+const isRequestAllowed = (
```

Comment:
I still need unit tests for this. You wrote unit tests for the template handler but this doesn't provide adequate coverage for this file. We need unit tests specifically for the isRequestAllowed middleware